Microgame/ice path/add assets

Alexyzyz opened the pull request:
[IcePath] Add assets

Alexyzyz committed:
[IcePath] Add assets

-Restructured the code.
-Added MapData ScriptableObject.
-The game can now have up to 3 Waka obstacles.
-The camera now zooms in on the ice cream before showing the whole map.
-Added current sprites.
-Added current music and sound effects.

Alexyzyz committed:
[IcePath] Victory state

-On victory, the camera now zooms in on Cirno with the victory animation.
-Tweaked Waka movement.

Gman8r reviewed (changes requested):
Needs the asset/camera work mentioned in discord

Alexyzyz committed:
[IcePath] Asset tweaks

-Reworked the sprites.
-Added some animations.
-Note: having trouble making Waka leap properly...

Alexyzyz committed:
[IcePath] Asset tweaks 2

-Adjusted background and new Waka animation.

Alexyzyz committed:
[IcePath] General tweaks

-No more framerate dependent movements.
-Ice cream and scenes for all three difficulties.
-Definitely not final maps added in.
* Some maps have variations for randomized finish tile locations which makes them suffer from being selected most of the time. Will tweak map selection code or resort to non-randomized finish tile map designs.

Alexyzyz committed:
[IcePath] General tweaks 2

-Maps can now have indefinite finish tiles (finish tiles can be randomly selected from up to 9 possible tiles).
-Updated Cirno's hit particle (every art asset in the game is now original).
-Game currently has 5-6 maps per difficulty. Next branch might be all about map design and reviews.

Alexyzyz committed:
[IcePath] General tweaks 3

-Cirno is no longer able to move diagonally.
-Adjustable Waka leap duration.

Gman8r reviewed (changes requested):
Implement changes about spritesheets, waka/rocks, and lilypads from Discord

Alexyzyz committed:
[IcePath] Requested changes

-Compressed tile sprites into a single spritesheet.
-Removed rock layer of Waka's idle tile; adjusted Waka's sprite accordingly.
-Placeholder maps.
* Maps are now 12x9 tiles in size.
